From 2dc4c672fd9db0d5e8945245ba0d62870e1151e4 Mon Sep 17 00:00:00 2001
From: Mamoru TASAKA <mtasaka@fedoraproject.org>
Date: Fri, 30 May 2014 00:05:14 +0900
Subject: [PATCH 5/6] Remove GLib deprecated warnings

g_type_init():
https://developer.gnome.org/gobject/stable/gobject-Type-Information.html#g-type-init
says that this call can simply be removed since 2.36.0, so wrap this with
GLIB_CHECK_VERSION .

gdk_pixbuf_unref():
https://developer.gnome.org/gdk-pixbuf/stable/gdk-pixbuf-Reference-Counting-and-Memory-Mangement.html#gdk-pixbuf-unref
says that this is deprecated since 2.0.0 and can simply be replaced with
g_object_unref(). As configure.in already requires pkgconfig gmodule-2.0 >= 2.0.0,
we can use g_object_unref().
---
 driver/xscreensaver-getimage.c |  2 ++
 hacks/glx/xpm-ximage.c         |  2 ++
 hacks/webcollage-helper.c      | 10 ++++++----
 hacks/xpm-pixmap.c             |  2 ++
 4 files changed, 12 insertions(+), 4 deletions(-)

diff --git a/driver/xscreensaver-getimage.c b/driver/xscreensaver-getimage.c
index 598a913..f22504e 100644
--- a/driver/xscreensaver-getimage.c
+++ b/driver/xscreensaver-getimage.c
@@ -381,7 +381,9 @@ read_file_gdk (Screen *screen, Window window, Drawable drawable,
 
   gdk_pixbuf_xlib_init_with_depth (dpy, screen_number (screen), win_depth);
 # ifdef HAVE_GTK2
+# if !GLIB_CHECK_VERSION(2, 36 ,0)
   g_type_init();
+# endif
 # else  /* !HAVE_GTK2 */
   xlib_rgb_init (dpy, screen);
 # endif /* !HAVE_GTK2 */
diff --git a/hacks/glx/xpm-ximage.c b/hacks/glx/xpm-ximage.c
index 4091c70..ea75d4f 100644
--- a/hacks/glx/xpm-ximage.c
+++ b/hacks/glx/xpm-ximage.c
@@ -78,8 +78,10 @@ xpm_to_ximage_1 (Display *dpy, Visual *visual, Colormap cmap,
   if (!initted)
     {
 #ifdef HAVE_GTK2
+#if !GLIB_CHECK_VERSION(2, 36 ,0)
       g_type_init ();
 #endif
+#endif
       gdk_pixbuf_xlib_init (dpy, DefaultScreen (dpy));
       xlib_rgb_init (dpy, DefaultScreenOfDisplay (dpy));
       initted = 1;
diff --git a/hacks/webcollage-helper.c b/hacks/webcollage-helper.c
index e6e2da7..673e8b4 100644
--- a/hacks/webcollage-helper.c
+++ b/hacks/webcollage-helper.c
@@ -85,7 +85,7 @@ bevel_image (GdkPixbuf **pbP, int bevel_pct,
   if (! gdk_pixbuf_get_has_alpha (pb))
     {
       GdkPixbuf *pb2 = gdk_pixbuf_add_alpha (pb, FALSE, 0, 0, 0);
-      gdk_pixbuf_unref (pb);
+      g_object_unref (pb);
       pb = pb2;
     }
 
@@ -205,7 +205,7 @@ paste (const char *paste_file,
       int new_h = paste_h * from_scale;
       GdkPixbuf *new_pb = gdk_pixbuf_scale_simple (paste_pb, new_w, new_h,
                                                    GDK_INTERP_HYPER);
-      gdk_pixbuf_unref (paste_pb);
+      g_object_unref (paste_pb);
       paste_pb = new_pb;
       paste_w = gdk_pixbuf_get_width (paste_pb);
       paste_h = gdk_pixbuf_get_height (paste_pb);
@@ -323,9 +323,9 @@ paste (const char *paste_file,
     fprintf (stderr, "%s: pasted %dx%d from %d,%d to %d,%d\n",
              progname, paste_w, paste_h, from_x, from_y, to_x, to_y);
 
-  gdk_pixbuf_unref (paste_pb);
+  g_object_unref (paste_pb);
   write_pixbuf (base_pb, base_file);
-  gdk_pixbuf_unref (base_pb);
+  g_object_unref (base_pb);
 }
 
 
@@ -491,7 +491,9 @@ main (int argc, char **argv)
   if (h < 0) usage();
 
 #ifdef HAVE_GTK2
+#if !GLIB_CHECK_VERSION(2, 36 ,0)
   g_type_init ();
+#endif
 #endif /* HAVE_GTK2 */
 
   paste (paste_file, base_file,
diff --git a/hacks/xpm-pixmap.c b/hacks/xpm-pixmap.c
index 9493e09..5e13a18 100644
--- a/hacks/xpm-pixmap.c
+++ b/hacks/xpm-pixmap.c
@@ -61,7 +61,9 @@ xpm_to_pixmap_1 (Display *dpy, Window window,
   if (!initted)
     {
 #ifdef HAVE_GTK2
+#if !GLIB_CHECK_VERSION(2, 36 ,0)
       g_type_init ();
+#endif
 #endif /* HAVE_GTK2 */
       gdk_pixbuf_xlib_init (dpy, screen_number (xgwa.screen));
       xlib_rgb_init (dpy, xgwa.screen);
-- 
1.9.3

